### PR TITLE
[FEATURE] Added REST API Call to Retrieve Last Order Information so t…

### PR DESCRIPTION
**Pull request by lewisvoncken**
…he Order Increment can be used in PWA

### Description (*)

When you place an order it should return the Increment Id and not the entity id

### Before changes
image

### After Changes
image

### Related PWA PR
https://github.com/magento-research/pwa-studio/pull/649

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @lewisvoncken. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Quote/Model/QuoteManagement.php [hunk 1]
```diff
@@ -377,7 +377,7 @@ public function placeOrder($cartId, PaymentInterface $paymentMethod = null)
         $this->checkoutSession->setLastOrderStatus($order->getStatus());
 
         $this->eventManager->dispatch('checkout_submit_all_after', ['order' => $order, 'quote' => $quote]);
-        return $order->getId();
+        return $order->getIncrementId();
```
Comment (orlangur) [inline app/code/Magento/Quote/Model/QuoteManagement.php]: This change is incorrect: https://github.com/magento/magento2/blob/65a8c75ba70bcb446865ce183e9d5b3c020d51ed/app/code/Magento/Quote/Api/CartManagementInterface.php#L64

Please apply a fix in a proper place.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by lewisvoncken** [comment]
@orlangur 
Thanks for the Feedback! I have created a separate API call to retrieve the Last Order Information, which is used in the pr https://github.com/magento-research/pwa-studio/pull/649

**Comment by orlangur** [comment]
@antonkril @buskamuza please review new API addition.

Shouldn't we retrieve OrderID and then obtain IncrementID with a separate API call instead of this?

**Comment by lewisvoncken** [comment]
@orlangur That was my first solution but when you have guest order you are not authorised to retrieve the order information.

**Comment by nuzil** [comment]
Hi here, if we are talking about creating REST API call for PWA usecase, I may think its a wrong approach. As I know main idea of PWA in the future is to be fully dependent from GraphQL API and not to REST. My thoughts that we need it on only in GraphQL, or in this case in both implementations.

**Comment by nuzil** [comment]
@zetlen can you maybe add something? or I'm wrong here?

**Comment by Vinai** [comment]
I would like it if the naming of the classes would be a bit more descriptive.

For one, there is no management capability involved, the class is only about querying information.
I suggest dropping the `Management` from the name.

And since the use case is coupled to the current session, I think the name `LastOrderInformationInterface` is lacking.
Just looking at the interface, I expect to be able to retrieve the last order information for the currently logged in customer, regardless if the last order was placed during the current session or in a previous session. 

So how about naming the API endpoint and class  `CurrentSessionLastOrderInformationInterface`?

**Comment by zetlen** [comment]
@nuzil is right; any customer-facing use cases should be implemented in GraphQL from now on. @lewisvoncken Thanks for proactively opening this issue! We need to rephrase it and open it in https://github.com/magento/graphql-ce/ instead.

**Comment by lewisvoncken** [comment]
> @nuzil is right; any customer-facing use cases should be implemented in GraphQL from now on. @lewisvoncken Thanks for proactively opening this issue! We need to rephrase it and open it in https://github.com/magento/graphql-ce/ instead.

@zetlen , @nuzil , @Vinai and @orlangur 
I have just ported it to GraphQL but I haven't implemented it in PWA Studio do you have any advice how to do so?
https://github.com/magento/graphql-ce/pull/298

**Comment by lewisvoncken** [comment]
@zetlen , @nuzil , @Vinai and @orlangur

I think this PR can be closed because the new one is created.

**Comment by nuzil** [comment]
@lewisvoncken as I see from PWA pull request update you already found how? :-)

**Pull request closed without merging**